src/bp/core/repositories/conversations.ts

```diff
@@ -0,0 +1,238 @@
+import * as sdk from 'botpress/sdk'
+import Database from 'core/database'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import uuid from 'uuid'
+
+const conversationsTable = 'web_conversations'
+const messageTable = 'web_messages'
+
+@injectable()
+export class ConversationsRepository {
+  constructor(@inject(TYPES.Database) private database: Database) {}
+
+  async getOrCreateConversation(
+    botId: string,
+    userId: string,
+    opt: { lifetime?: string; isUserCreated?: boolean } = { lifetime: '6 hours', isUserCreated: false }
+  ): Promise<any> {
+    const lessThanLifetime = this.database.knex.date.isAfter(
+      'last_heard_on',
+      moment()
+        .subtract(ms(opt.lifetime), 'ms')
+        .toDate()
+    )
+
+    const conversation = await this.database
```

Comment:
You could check the `isUserCreated` option first and skip this call if its false as the conversation is probably not created if the user is not created. wht do you think ?